### PR TITLE
refactor(files): Adjust some Typescript code in virtual files list

### PR DESCRIPTION
**Pull request by susnux**
## Summary
- Correctly cast table ref to fix TypeScript errors (I think this also makes it a bit better readable :) )
- Refactor the `filesListWidth` mixin
  Use `defineComponent` to properly inherit typings, also add the `$resizeOberserver` through the setup to the component instance.
  `filesListWidth` is now a number which defaults to 0, making compares like `this.fileListWidth < 768` valid in Typescript (there are a lot of Type error in other places because of this).
- Properly add type information to `FilesListTableFooter`
  - Add correct type for `nodes` prop.
  - Use `defineComponent` to properly infer Typescript information
  - Correct usage of nullish coalescing operator
- Make `currentView` a prop of `FilesListTableFooter`
  We already have this as a required prop in `FilesListVirtual`, so passing it to the footer is straight forward and removed need of computed value.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by susnux — COMMENTED** [review]
Added some hints about the problems I try to solve

File: apps/files/src/mixins/filesListWidth.ts [hunk 1]
```diff
@@ -3,26 +3,31 @@
...
+			filesListWidth: 0,
 		}
 	},
+
 	mounted() {
 		const fileListEl = document.querySelector('#app-content-vue')
-		this.filesListWidth = fileListEl?.clientWidth ?? null
+		this.filesListWidth = fileListEl?.clientWidth ?? 0
 
+		// @ts-expect-error The resize observer is just now attached to the object
```
Comment (susnux) [inline apps/files/src/mixins/filesListWidth.ts:19]: We can not use `setup` in mixins, so we can not attach it properly with type information.
Using `data` also feels wrong, it should not be made reactive.

Comment (ShGKme) [reply apps/files/src/mixins/filesListWidth.ts:19]: This mixin is quite small and only defines one property. Maybe we can rewrite it to a composable then?

**Review by ShGKme — APPROVED** [review]

**Comment by ShGKme** [comment]
Thanks for detailed description about all the changes!

File: apps/files/src/mixins/filesListWidth.ts (hunk 1 above)
Comment (susnux) [reply apps/files/src/mixins/filesListWidth.ts:19]: Yes I also was thinking about that.
Also a composable for `dir` and `fileid` makes sense as we use it in many places and just duplicate logic there.

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**